### ISSUE TITLE
Compatible with JF BAE146

### ISSUE THREAD

**Issue by bagolu** [dementedmonkey/cj4-mcdu#23]
not an issue, but with the same two files in 
  html_ui/Pages/VCockpit/Instruments/Airliners/JF_146/FMC
the mod works with the Justflight BAE146 (it uses WT cj4 fmc)

**Comment by dementedmonkey** [comment]
I could try putting together another mod, but I don't have the 146 and can't test it.

Would you be willing to test this out for me?

**Comment by 1l2p-dev** [comment]
Hi, thanks for this feedback.

I made it work by simply renaming the 'manifest.json' filenames in a new mod (and matching the BAE146 paths).
I'm using SPAD.neXt and also had to replace '□' by '_' chars in "WT_FMC_Renderer.js" to have it handled by SPAD (`text.replace(/□/g, "_");`). This should definitely be optimized as it was just a test.

Do not hesitate to ask if you need help with further testing.

It's a real pleasure to have your MOD working on CDU hardware.

[z-dementedmonkey-jf_146-mcdu.zip](https://github.com/dementedmonkey/cj4-mcdu/files/9078770/z-dementedmonkey-jf_146-mcdu.zip)

**Comment by linmar22** [comment]
+1 for this being a fork or just a separate build in the releases. Willing to get the 146 and test for you @dementedmonkey.

**Comment by taipan303** [comment]
I confirm it works, there's quite a few people using it with the 146 now.
It also enables CDU access for the Cockpit Master CDU (hardware addon).

JustFlight are creating a new FMS for the BAE146 as part of the Avro development, but they have said the user will have the choice to keep the original WT FMS as an option.

**Comment by taipan303** [comment]
Testing this new version for the BAE146 from the 0.3.0 changes. Note that it only has 3 javascript files instead of 4 because the initref page of the BAE146 is different and shouldn't be overwritten. The initref page doesn't need to change for this to work (according to my testing). 
To use both the CJ4 and BAE146 with the CM CDU you need both mods. 

It doesn't include the web server, use the same server or use SPAD.

[bae146-mcdu-0.3.0.zip](https://github.com/dementedmonkey/cj4-mcdu/files/9998809/bae146-mcdu-0.3.0.zip)

I have only done some testing on the ground, won't have time for a flight for a while.
Let me know if any issues or fixes

**Comment by dementedmonkey** [comment]
You're right, the initref page update eliminates some flickering and fixes a bug where the previous waypoint distance wasn't defined.   These were just issues in the base CJ4 package, not really specific to, or required for, the MCDU mod.

**Comment by radiowar69** [comment]
I keep getting this error when starting the server. And I apologize I now little more than how to install things. "Iconv-lite warning: decode()-ing strings is deprecated. Refer to https://github.com/ashtuchkin/iconv-lite/wiki/Use-Buffers-when-decoding"

Any thoughts on why I would be getting this error?

**Comment by SimPitMaker** [comment]
Does the bae146 mod work for the RJ Professional?

**Comment by dementedmonkey** [comment]
I can't support a plane I don't have, and the 146/RJ pro isn't something I was planning on buying.
